lib/vsc/install/shared_setup.py

```diff
@@ -331,7 +332,15 @@ def __init__(self, *args, **kwargs):
                 kwargs['output'] = xmlrunner_output
                 xmlrunner.XMLTestRunner.__init__(self, *args, **kwargs)
 
-        main_orig = unittest.main
+        # old setuptools
+        main_name = 'main'
+        main_orig = getattr(setuptools.command.test, main_name, None)
+        if main_orig is None:
+            # newer setuptools (e.g. 0.18)
+            main_name = 'unittest_main'
+            main_orig = getattr(setuptools.command.test, main_name, None)
+        if main_orig is None:
+            raise Exception('monkey patching XmlRunner failed')
```

Comment:
it's not about readability, it's about avoid duplicatie code, and making it trivial to extend when setuptools changes their mind in a future version (they will, trust me ;-))